src/currentWeather/SearchHistory.tsx

```diff
@@ -0,0 +1,40 @@
+import "./SearchHistory.css";
+
+type SearchHistoryProps = {
+  searchHistory: string[];
+  onSetLocation: React.Dispatch<React.SetStateAction<string>>;
+  onClick: (location: string) => Promise<void>;
+  onSetIsActive: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const SearchHistory = ({
+  searchHistory,
+  onSetLocation,
+  onClick,
+  onSetIsActive,
+}: SearchHistoryProps) => {
+  function handleClick(event: React.MouseEvent<HTMLElement>) {
+    onClick((event.target as HTMLInputElement).value);
+    onSetIsActive(false);
```

Comment:
You don't need these props. Just call `onClick` and make this stuff in parent component. This component doesn't care about some active or location state.

you should do it in parent, where this things make sense, not here, where these things are not used anywhere